src/java/org/dom4j/io/XMPPPacketReader.java

```diff
@@ -364,15 +375,13 @@ public Document parseDocument() throws DocumentException, IOException, XmlPullPa
                 }
                 case XmlPullParser.START_TAG: {
                     QName qname = (pp.getPrefix() == null) ? df.createQName(pp.getName(), pp.getNamespace()) : df.createQName(pp.getName(), pp.getPrefix(), pp.getNamespace());
-                    Element newElement = null;
-                    // Do not include the namespace if this is the start tag of a new packet
-                    // This avoids including "jabber:client", "jabber:server" or
-                    // "jabber:component:accept"
-                    if ("jabber:client".equals(qname.getNamespaceURI()) ||
-                            "jabber:server".equals(qname.getNamespaceURI()) ||
-                            "jabber:connectionmanager".equals(qname.getNamespaceURI()) ||
-                            "jabber:component:accept".equals(qname.getNamespaceURI()) ||
-                            "http://jabber.org/protocol/httpbind".equals(qname.getNamespaceURI())) {

```

Comment:
Absolutely prefer to push as-is, but it might be worth checking the content namespace is consistent and correct, given that otherwise we won't detect when someone sends jabber:server over S2S, for example.

Reply:
(aren't you supposed to send jabber:server over s2s? You ment jabber:client, right?)

I'm not sure what you mean here. Yes, we might want to check, inbound, that the namespace matches the connection type (although I think that RFC6120 doesn't require that and even allows for other namespaces to be used - which would all be terribly confusing). On the outbound stream, we define the default namespace in our code. As all stanzas are stripped of their default namespace _on the stanza level_ (not in their child elements), things should be OK?